File: ud01-introduccion-entorno-servidor/02-ejemplos/hello-server/src/main/java/edu/iesrafaelalberti/dwes/helloserver/HelloController.java
package edu.iesrafaelalberti.dwes.helloserver;

import java.util.Map;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

    @GetMapping(value = "/", produces = MediaType.TEXT_HTML_VALUE)
    String home() {
        return """
                <!doctype html>
                <html lang="es">
                  <head><meta charset="utf-8"><title>Hello Server</title></head>
                  <body><h1>Hola desde el servidor</h1></body>
                </html>
                """;
    }

    @GetMapping("/api/hello")
    Map<String, String> hello() {
        return Map.of("message", "Hola desde el servidor");
    }

    @GetMapping("/health")
    Map<String, String> health() {
        return Map.of("status", "UP");
    }
}

File: ud02-api-rest/ud02a-spring-boot/02-ejemplos/SpngBoot-4/demoSpBt4Kt/src/main/kotlin/com/example/demospbt4kt/controller/HelloController.kt
package com.example.demospbt4kt.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController {
    @GetMapping
    fun hello(): String {
        if (System.getProperty("org.graalvm.nativeimage.imagecode") != null) {
            return "Hola desde Spring Boot con Kotlin nativo!"
        }
        return "Hola desde Spring Boot con Kotlin en JVM!"
    }
}